astropy/units/quantity_helper/function_helpers.py

```diff
@@ -497,6 +501,63 @@ def arange_impl(*args, start=None, stop=None, step=None, dtype=None, device=None
     return np.arange(**kwargs), out_unit, None
 
 
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C'):
+        return empty_impl(shape, dtype, order)
+else:
+    @dispatched_function
+    def empty(shape, dtype=None, order='C', *, device=None):
+        return empty_impl(shape, dtype, order, device=device)
+
+def empty_impl(shape, dtype=None, order='C', *, device=None):
+    if NUMPY_LT_2_0:
+        ret_arr = np.empty(shape, dtype, order)
+    else:
+        ret_arr = np.empty(shape, dtype, order, device=device)
+    return ret_arr, _UnitFromLikeArg, None
+
+
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def ones(shape, dtype=None, order='C'):
+        return ones_impl(shape, dtype, order)
+else:
+    @dispatched_function
+    def ones(shape, dtype=None, order='C', *, device=None):
+        return ones_impl(shape, dtype, order, device=device)
+
+def ones_impl(shape, dtype=None, order='C', *, device=None):
+    if NUMPY_LT_2_0:
+        ret_arr = np.ones(shape, dtype, order)
+    else:
+        ret_arr = np.ones(shape, dtype, order, device=device)
+    return ret_arr, _UnitFromLikeArg, None
+
+@dispatched_function
+def zeros(shape, dtype=None, order='C'):
```

Comment:
Does `zeros` not have the `device` argument?

Reply:
It doesn't. Otherwise we'd be seeing at least one test fail, but I admit I too was surprised that its signature differs from that of ones. It's probably either an oversight or reflects that ones is part of the Array API and zeros isn't.